modelkit/core/library.py

```diff
@@ -145,12 +148,18 @@ def get(self, name):
         """
         try:
             if not self._lazy_loading:
```

Comment:
To avoid redundancies, we can write this one such as:
```python
            if self._lazy_loading:
                if name not in self.models:
                    self._load(name)
                if not self.models[name]._loaded:
                    self.models[name].load()
            m = self.models[name]
            if model_type and not isinstance(m, model_type):
                raise ValueError(f"Model `{m}` is not an instance of {model_type}")
            return cast(T, self.models[name])
```
As discussed, this section seems rather counter-intuitive even though it behaves as intented.

Reply:
Thanks for the comment, I added some comments in an attempt to clarify what happens.

Reply:
I integrated your suggestion